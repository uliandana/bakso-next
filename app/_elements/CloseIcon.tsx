import React from 'react';

type Props = {
  className?: string,
};

export default function CloseIcon(props: Readonly<Props>) {
  const { className } = props;
  return (
    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className={className}>
      <path strokeLinecap="round" strokeLinejoin="round" d="M6 18 18 6M6 6l12 12" />
    </svg>
  );
}
