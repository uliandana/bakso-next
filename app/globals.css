@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --foreground-rgb: 255, 255, 255;
  --background-start-rgb: 188,55,54;
  --background-end-rgb: 188,55,54;
  font-size: calc(768px * 16 / 375);
}

@media (prefers-color-scheme: dark) {
  :root {
    --foreground-rgb: 255, 255, 255;
    --background-start-rgb: 0, 0, 0;
    --background-end-rgb: 0, 0, 0;
  }
}

@layer utilities {
  .card-flip {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(0);
    transition: transform 0.5s ease-in-out;
  }
  .card-flipped {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(180deg);
    transition: transform 0.5s ease-in-out;
  }
  .silhouette {
    filter: brightness(0) drop-shadow(0.0625rem 0.125rem 0.25rem rgb(var(--background-start-rgb)));
  }
}

body {
  color: rgb(var(--foreground-rgb));
  background: linear-gradient(
      to bottom,
      transparent,
      rgb(var(--background-end-rgb))
    )
    rgb(var(--background-start-rgb));
}

* {
  animation: fade-in 0.5s ease-in-out;
}

@keyframes loading {
  to { transform: translateX(100%); }
}

@keyframes fade-in {
  from { opacity: 0; }
  to { opacity: 1; }
}

@media only screen and (max-width: 1440px) {
  :root {
    font-size: calc(100vw * 16 / 1440);
  }
}

@media only screen and (max-width: 768px) {
  :root {
    font-size: calc(100vw * 16 / 768);
  }
}
