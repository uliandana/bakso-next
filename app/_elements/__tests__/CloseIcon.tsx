import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { createRenderer } from 'react-test-renderer/shallow';
import CloseIcon from '../CloseIcon';

describe('./_elements/CloseIcon', () => {
  test('renders', () => {
    const tree = createRenderer().render(<CloseIcon />);
    expect(tree).toMatchSnapshot();
  });
});
