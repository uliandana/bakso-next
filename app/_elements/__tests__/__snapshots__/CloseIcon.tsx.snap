// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./_elements/CloseIcon renders 1`] = `
<svg
  fill="none"
  stroke="currentColor"
  strokeWidth={1.5}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M6 18 18 6M6 6l12 12"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</svg>
`;
